### DISCUSSION TITLE
Is there any way to flash from USB?

### DISCUSSION THREAD

**Discussion by Allochtone** [bigtreetech/BIGTREETECH-TouchScreenFirmware#1990] [Q&A]
Is there a simple way to do this? I don't mind diving in the firmware to tweak some things here and there, but if you tell me I need to reinvent the wheel, I'll reconsider. Since both SD card and USB stick have the same format (FAT32), and both can be read in the original firmware, I do believe it is possible (maybe not easy).

Thank you!

**Comment by Mactastic1-5** [comment]
It can only be updated via SD Card.

**Comment by burneme** [comment]
This requires that the boot-loader is able to have a driver for USB to get access to a USB-stick during startup. Boot loaders are kept small not to waste too much memory. A SD-Card is "quite easy" to access by SPI-Interface, so it is more reasonable to offer just the SD as the offical way to update.
